### PR TITLE
auxiliary atom feed - implementing lambda to upload auxiliary atoms.

### PR DESCRIPTION
**Pull request by LATaylor-guardian**
@clloyd Would be great if you could take a look at this.

I have created the following buckets in S3 in your account:
 - auxiliary-atom-feed-dist
 - auxiliary-atom-feed-config

Next steps: 
 - [x] Test new flex api endpoint (and deploy to master if successful)
 - [x] Create kinesis stream
 - [ ] Create Lambda
 - [ ] Hook Lambda's up to kinesis stream.
 - [ ] Test end to end.
 - [ ] Repeat for PROD.

@clloyd Do you have a cloudformation somewhere where you want me to script the creation of the kinesis streams, or do you just do them in the console? The same will go for permissions to write to this stream cross accounts.

### CODE REVIEW

File: src/main/scala/com/gu/flexible/Config.scala [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+import java.util.Properties
+
+import com.gu.flexible.services.S3
+
+import scala.util.Try
+
+class Config(functionName: String) {
+
+  val isProd = Try(functionName.toLowerCase.contains("-prod")).getOrElse(false)
```
Comment (oilnam) [inline src/main/scala/com/gu/flexible/Config.scala]: Haha nice 😄

File: src/main/scala/com/gu/flexible/Lambda.scala [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+class Lambda {
+
+  def recordHandler(event: KinesisEvent, context: Context) {
+
+    val config = new Config(context.getFunctionName)
+
+    event.getRecords.iterator.asScala.foreach { kinesisEventRecord =>
+
+      val data = kinesisEventRecord.getKinesis.getData
+      val op = ThriftDeserializer.deserializeEvent[AuxiliaryAtomEvent](data, codec = AuxiliaryAtomEvent)
```
Comment (oilnam) [inline src/main/scala/com/gu/flexible/Lambda.scala]: Very very minor but that does `op` stands for?

Comment (LATaylor-guardian) [inline src/main/scala/com/gu/flexible/Lambda.scala]: I'll change it - `deserializationAttempt`

**Comment by oilnam** [comment]
👍

**Pull request merged**